rasa/shared/importers/importer.py

```diff
@@ -488,12 +488,19 @@ async def get_nlu_data(self, language: Optional[Text] = "en") -> TrainingData:
     async def _additional_training_data_from_stories(self) -> TrainingData:
         stories = await self.get_stories()
 
-        additional_messages_from_stories = []
-        for story_step in stories.story_steps:
-            for event in story_step.events:
-                message = _message_from_conversation_event(event)
-                if message:
-                    additional_messages_from_stories.append(message)
+        utterances, actions = _unique_events_from_stories(stories)
+
+        # Sort events to guarantee deterministic behavior and to avoid that the NLU
+        # model has to be retrained due to changes in the event order within
+        # the stories.
+        sorted_utterances = sorted(
+            list(utterances), key=lambda user: user.intent_name or user.text
+        )
+        sorted_actions = sorted(list(actions), key=lambda action: action.action_name)
```

Comment:
why are we not using `action_text` here?

Reply:
I was under the impression that always `action_name==action_text` . Added it now 👍